### PR TITLE
Move comments fetch from the CommentsProvider to the CommentList

### PR DESCRIPTION
**Pull request by sergous**
https://github.com/dtpstat/website/issues/48

### CODE REVIEW

File: src/providers/comments-provider.tsx [hunk 1]
```diff
@@ -20,18 +18,6 @@ export const CommentsProvider: React.VoidFunctionComponent<{
...
-  React.useEffect(() => {
-    const initFetchComments = async () => {
-      const initialComments = await fetchComments(
-        window.location.origin,
-        accidentId,
-      );
-      setComments(initialComments);
-    };
-    void initFetchComments();
-  }, [accidentId]);
```
Comment (kachkaev) [inline src/providers/comments-provider.tsx:34]: I believe we can achieve the same result while keeping our fetch function where it was. E.g.:

```ts
  React.useEffect(() => {
    let effectIsStale = false;
    
    const fetchAndSetComments = async () => {
      const comments = await fetchComments(
        window.location.origin,
        accidentId,
      );

      if (!effectIsStale) {
        setComments(comments);
      }
    };
    
    void fetchAndSetComments();

    return () => {
      effectIsState = true;
    }
  }, [accidentId]);
```

If we move fetching to the list of comments, the value of provider diminishes and it no longer does what it ‘says on the tin’.

Comment (sergous) [reply src/providers/comments-provider.tsx:34]: This move for preventing memory leak (https://github.com/dtpstat/website/issues/48)

Comment (kachkaev) [reply src/providers/comments-provider.tsx:34]:
```
if (!effectIsStale) { setComments(comments); }
```

should prevent it

Comment (kachkaev) [reply src/providers/comments-provider.tsx:34]: You’ve done pretty much the same change, which is correct. I’m just suggesting to keep `useEffect` in the original component. If you still see a warning, let me know and I’ll play with the code locally 👀

Comment (sergous) [reply src/providers/comments-provider.tsx]: Got it, moved.

**Review by kachkaev — APPROVED** [review]

**Pull request merged**